index.d.ts

```diff
@@ -0,0 +1,35 @@
+// Type definitions for node-amqp-bus 5.1.0
+// Project: node-amqp-bus
+// Definitions by: Chauffeur Privé
+// TypeScript Version: 3.0.1
+
+/// <reference types="node" />
+
+import { EventEmitter } from 'events';
+import { Connection, Channel } from 'amqplib';
+
+export default Bus;
+
+declare class Bus {
+  static createClient(url: string, options?: Bus.BusOptions): BusClient;
+}
+
+export class BusClient extends EventEmitter {
+  channel: Channel;
+  connection: Connection;
+
+  publish(exchange: string, routingKey: string, message: Object, options?: Bus.PublishOptions): Boolean;
+}
+
+declare namespace Bus {
+  export interface BusOptions {
+    heartbeat?: number;
+    useConfirmChannel?: Boolean;
+  }
+
+  export interface PublishOptions {
+    // publish options type not exported in the amqp types
+    // https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3ea5ad1/types/amqplib/properties.d.ts#L108
+    [propName: string]: any;
```

Comment:
I thought about it at first but we don't really use these options to publish messages today so I thought it was better to avoid any friction in case of update. Is it ok for you?